### PR TITLE
chore: add issue and PR templates

### PR DESCRIPTION
**Pull request by frederickfogerty**
## Description
PR and issues templates help guide users to create better issues and PRs, which help contributors maintain the repo.

### Documentation

- [x] The PR title is in the conventional commit format: e.g. `chore(readme): updated documentation for ___`
- [x] Add your info to the [contributors](#packagejson-contributors) array in package.json!

[issues]: ../../issues

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+name: Bug report
+about: Create a report to help us improve
+
+---
+
+**Checklist:**
+
+- [ ] I have checked stackoverflow for this bug and there are no answers or the question does not exist
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I added this checklist item as I'd remembered reading [this section](https://github.com/imgix/drift/blob/b2acc8e4e0e834d80a40d6c12643c473fe17f071/CONTRIBUTING.md#using-the-issue-tracker) in the `CONTRIBUTING.md` file for this repository. I now realised that this was a misinterpretation of what that meant, so I'll remove this item.

[the remark this answers is not in the document]
Comment (frederickfogerty) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Re the checklist - yes, it was added due to experience. I remember it being a common pattern among open-source repositories that I've seen. This, along with the psychological influence mentioned, is the reason for structuring it this way.

There have been a few times that I've created a bug report/question/PR and then had someone point out that it's a duplicate or that I hadn't followed a particular process, even when there was a sentence indicating that I should do so. I think the majority of people will do the same as me - skim over the instructions. If instead it was presented in a (short) checklist, I think people would be less likely to skim over it.

Is there any way we can reduce how passive-aggressive it is? Is it the wording of the checklist items, or the fact that a checklist is being used, that is causing it to be passive-aggressive? I'd rather err on the side of being welcoming and not passive-aggressive (or any type of aggression for that matter), so I'm happy to remove this if we can't remedy its passive-aggressive nature.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Information:**
+
+- drift version: [e.g. v1.0]
+
+**Desktop (please complete the following information):**
+
+- OS: [e.g. iOS]
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I haven't seen that site before, it looks super useful. I'll update this section.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+- [description](#description): please tell us about your pr
+- [checklist](#checklist): please review the checklist that is most closly related to your pr
+- [contributors list](#packagejson-contributors): you're one of us now, please add yourself to `package.json` and let the world know!
+
+The following sections provide more detail on each.
+
+**Improve this document**
+
+Please don't hesitate to [ask questions][issues] for clarification, or to [make suggestions][issues] (or a pull request) to improve this document.
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply .github/pull_request_template.md]: Sure does!

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+`description` should be a short description of the change, in past tense
+
+## package.json contributors
+
+**Add yourself!**
+
+When adding your information to the `contributors` array in package.json, please use the following format (provide your name at minimum, the other fields are optional):
+
+```
+full name <email@address.com> (https://website.com)
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply .github/pull_request_template.md]: I'm happy to allow both. What do you think?

File: .github/ISSUE_TEMPLATE/question.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+name: Question
+about: Ask a Question about the project
+
+---
+
+**Checklist:**
+
+- [ ] I have checked stackoverflow for this question and there are no answers or the question does not exist
```
Comment (frederickfogerty) [inline .github/ISSUE_TEMPLATE/question.md]: @jayeb Are you happy to keep this requirement here - I believe this is consistent with what's described in `CONTRIBUTING.md`

File: .github/pull_request_template.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jayeb) [reply .github/pull_request_template.md]: Both 👍

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jayeb) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I think it's the sentence structure itself that bothers me--something about checking boxes labelled "I have..." feels condescending. What if the header was changed to "**Before you submit:**" and the text was instead formatted as simple instructions?

``` markdown
**Before you submit:**

- [ ] Search through the issue list to make sure your bug or feature has not already been discussed.
```

This way we'd get the benefit of seeing a checkbox (which I agree, will stand out to someone skimming the document, great point), but ensure the submitter doesn't feel talked-down-to. What do you think?

File: .github/ISSUE_TEMPLATE/question.md (hunk 5 above)
Comment (jayeb) [reply .github/ISSUE_TEMPLATE/question.md]: I think I feel the same way--I don't want to point people out to another forum if they've already found their way here. I'd rather treat this repo as the canonical source for questions and answers about this project. If you disagree strongly, though, I could probably be convinced to leave it.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Sounds great to me, I'll update these.

File: .github/ISSUE_TEMPLATE/question.md (hunk 5 above)
Comment (frederickfogerty) [reply .github/ISSUE_TEMPLATE/question.md]: I don't feel strongly about this - the reason I brought it up is that I don't want to disagree with `CONTRIBUTING.md`. We're on the same page i.e. keep this repo the source for Q&A. I'll need to update the `CONTRIBUTING` to reflect this as well.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -4,23 +4,26 @@ about: Create a report to help us improve
...
-- [ ] I have searched through the issue list for related bugs or issues
+- [ ] Please search through the existing issues (both open AND closed) to see if your issue has been discussed before. Github issue search can be used for this: https://github.com/imgix/drift/issues?utf8=%E2%9C%93&q=is%3Aissue
+- [ ] Please ensure the problem has been isolated and reduced. This link explains more: http://css-tricks.com/6263-reduced-test-cases/
 
 **Describe the bug**
-A clear and concise description of what the bug is.
+A clear and concise description of what the bug is. Please strive to reach the **root problem** of your issue to avoid the XY problem. See more: https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem
 
 **To Reproduce**
+A bug is a _demonstrable problem_ that is scause by the code in the repository. Thus, the contributors need a way to reproduce your issue - if we can't reproduce your issue, we can't help you! Also, please be as detailed as possible!
```
Comment (jayeb) [inline .github/ISSUE_TEMPLATE/bug_report.md]: s/scause/caused, and let's switch the exclamation point in the last sentence here to a period.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+3.  If you've fixed a bug or added code that should be tested, add tests!
+4.  Ensure the test suite passes (`npm run test`). Tip: `npm run --watch` is helpful in development.
+5.  Format your code with [prettier](https://github.com/prettier/prettier) (`npm run format`). Don't worry too much if you haven't done this, we have a bot that will do this for you when you submit a PR.
+6.  Fill out the template below and delete everything above the line.
+
+**Learn more about contributing:** https://github.com/imgix/drift/blob/master/CONTRIBUTING.md
+
+## tldr
+
+There are three main goals in this document, depending on the nature of your pr:
```
Comment (jayeb) [inline .github/pull_request_template.md]: Ironically, this line still has a lowercase `pr`.

Comment (frederickfogerty) [reply .github/pull_request_template.md]: Argh, my search didn't pick this up!

**Comment by frederickfogerty** [comment]
Woo! Merging this, and then replicating across repos. Cheers for the constructive discussion @jayeb 🎉

**Pull request merged**